packedForest/src/forestTypes/unsupervisedForests/urerf/urerfTree.h

```diff
@@ -0,0 +1,267 @@
+#ifndef urerfTree_h
+#define urerfTree_h
+#include "../../../baseFunctions/fpBaseNode.h"
+#include "unprocessedURerFNode.h"
+#include <vector>
+#include <map>
+#include <assert.h>
+
+namespace fp{
+
+	template <typename T>
+		class urerfTree
+		{
+			protected:
+				float OOBAccuracy;
+				float correctOOB;
+				float totalOOB;
+				std::vector< fpBaseNode<T, std::vector<int> > > tree;
+				std::vector< unprocessedURerFNode<T> > nodeQueue;
+				std::vector< unprocessedURerFNode<T> > leafNodes;
+
+			public:
+				urerfTree() : OOBAccuracy(-1.0),correctOOB(0),totalOOB(0){}
+
+				void loadFirstNode(){
+					nodeQueue.emplace_back(fpSingleton::getSingleton().returnNumObservations());
+				}
+
+				inline bool shouldProcessNode(){
+					if(nodeQueue.back().returnNodeImpurity()<=0){
+						return false;
+					}
+					if(nodeQueue.back().returnInSampleSize() <= fpSingleton::getSingleton().returnMinParent()){
+						return false;
+					}
+                if(nodeQueue.back().returnDepth() >= fpSingleton::getSingleton().returnMaxDepth()){
+                          return false;
+                      }
+					return true;
+				}
+
+				inline float returnOOB(){
```

Comment:
What you have here seems outdated. I believe this should be 
```
inline std::vector<std::vector<int> > returnOOBvotes(){
					return indexAndVote;
				}
```

Reply:
checkOOB is not applicable for unsupervised random forests so I added an error message.